src/main/java/managers/PageObjectManager.java

```diff
@@ -1,18 +1,34 @@
 package managers;
 
+import base.BasePage;
+import exceptions.PageNotFoundException;
 import org.openqa.selenium.WebDriver;
 import pages.*;
 
 public class PageObjectManager {
     private WebDriver driver;
     private HomePage homePage;
+
     public PageObjectManager(WebDriver driver) {
         this.driver = driver;
     }
 
-    public HomePage getHomePage () {
-      return (homePage == null) ? homePage = new HomePage(this.driver) : homePage;
+    public HomePage getHomePage() {
+        return (homePage == null) ? homePage = new HomePage(this.driver) : homePage;
     }
 
+    public BasePage getPageByName(String pageName) throws PageNotFoundException {
+        BasePage page = null;
+
+        switch (pageName.toLowerCase()) {
+            case "home page":
```

Comment:
for bigger projects this might be a start of a an ugly switch case with a lot of cases. Do we really need this?

Reply:
Right now we keep it like that in almost every project, because it's used for cucumber steps like `User is on <page> page`, where <page> is an argument. Without a step that takes arguments we'd have to create a step for every page in a project
Do you have some better solution? We could discuss it with more people of course

Reply:
Maybe we can think about generics and not use switch case?

Reply:
https://www.baeldung.com/java-method-reflection